src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/SameNodeRetryPolicy.java

```diff
@@ -0,0 +1,100 @@
+package io.stargate.sgv2.jsonapi.service.cqldriver;
+
+import com.datastax.oss.driver.api.core.ConsistencyLevel;
+import com.datastax.oss.driver.api.core.connection.ClosedConnectionException;
+import com.datastax.oss.driver.api.core.connection.HeartbeatException;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.driver.api.core.retry.RetryDecision;
+import com.datastax.oss.driver.api.core.servererrors.CoordinatorException;
+import com.datastax.oss.driver.api.core.servererrors.ReadFailureException;
+import com.datastax.oss.driver.api.core.servererrors.WriteFailureException;
+import com.datastax.oss.driver.api.core.session.Request;
+import com.datastax.oss.driver.internal.core.retry.DefaultRetryPolicy;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A retry policy that retries to the same node in the query plan. DefaultRetryPolicy retries to the
```

Comment:
Could you add a brief mention why this is useful for our case and more generally due to the way proxies like CQLRouter work?

Reply:
Sure, added a comment in [this commit](https://github.com/stargate/jsonapi/pull/734/commits/3e05ad4110be60b6018ccc23c23f0cbbd8c8542e).